samples/cellular/nrf_cloud_rest_cell_location/prj.conf

```diff
@@ -38,7 +38,7 @@ CONFIG_MODEM_INFO=y
 CONFIG_MODEM_INFO_ADD_DEVICE=y
 CONFIG_MODEM_INFO_ADD_NETWORK=y
 CONFIG_MODEM_INFO_ADD_DATE_TIME=n
-CONFIG_MODEM_INFO_ADD_SIM=n
+CONFIG_MODEM_INFO_ADD_SIM=y
```

Comment:
Is this relevant to this PR?

Reply:
Changes to address Justin's comment here:
https://github.com/nrfconnect/sdk-nrf/pull/12034#discussion_r1294960739
did not build correctly without enabling this config to add the SIM info.